src/components/NotesMindMap/NotesMindMapComponent.js

```diff
@@ -63,31 +62,43 @@ export class NotesMindMapComponent extends Component {
     );
   }
 
+  convertToVisGraph() {
+    const notes = []
+    const edges = []
+
+    for (let i =0;i<this.props.notes.length;i++) notes.push(this.props.notes[i])
+    for (let i =0;i<this.props.edges.length;i++) edges.push(this.props.edges[i])
+
+    return {
+      nodes: [...this.props.notes], edges: [...this.props.edges]
+    }
+  }
+
   noteClickHandler = targetNoteId => {
     const { selectedNote } = this.props;
 
     // if clicking on the current note, do nothing.
     if (targetNoteId === selectedNote.id) return;
 
-    const rootNote = this.props.rootNote;
+    const notes = this.props.notes;
 
-    const targetNote = tree(rootNote).find(node => node.id === targetNoteId);
+    const targetNote = notes.find(note => note.id === targetNoteId);
 
     if (!targetNote) {
       throw new Error('noteClickHandler(): couldn\'t find targetNote: ', targetNoteId);
     }
 
-    this.props.changeSelectedNote(targetNote);
+    this.props.changeSelectedNote({ note: targetNote, edges: this.props.edges });
```

Comment:
Does `changeSelectedNote` requires info about `edges`?

Reply:
Passing to check if a note's children have been fetched.